validphys2/src/validphys/results.py

```diff
@@ -1098,6 +1153,52 @@ def fits_chi2_table(
     return res
 
 
+@table
+def fits_nsigma_table(
+    fits_total_chi2_data,
+    fits_datasets_nsigma_table,
+    fits_groups_nsigma_table,
+    show_total: bool = False,
+):
+    """Show the nsigma of each and number of points of each dataset and experiment for each fit,
+    computed with the theory corresponding to the fit.
+    Datasets that are not included in one of the fit appear as `NaN`
+    """
+    lvs = fits_groups_nsigma_table.index
+    # The explicit call to list is because pandas gets confused otherwise
+    expanded_index = pd.MultiIndex.from_product((list(lvs), ["Total"]))
+    edf = fits_groups_nsigma_table.set_index(expanded_index)
+    ddf = fits_datasets_nsigma_table
+    dfs = []
+    # TODO: Better way to do the merge preserving the order?
```

Comment:
```suggestion
    # Merge the columns from groups and per-dataset preserving the order coming from `fit_groups`
```

Please check that this comment is correct (if not change it accordingly).

Reply:
this comment was already in `fits_chi2_table`. It's there because, I mainly just copied that function

Reply:
If you want I can change the comment to your suggestion or remove it completely

Reply:
both work